trie/zk_trie_proof_test.go

```diff
@@ -195,3 +195,40 @@ func randomZktrie(t *testing.T, n int) (*ZkTrie, map[string]*kv) {
 
 	return tr, vals
 }
+
+// Tests that new "proof with deletion" feature
+func TestProofWithDeletion(t *testing.T) {
+	tr, _ := NewZkTrie(common.Hash{}, NewZktrieDatabase((memorydb.New())))
+	mt := &zkTrieImplTestWrapper{tr.Tree()}
+	key1 := bytes.Repeat([]byte("k"), 32)
+	key2 := bytes.Repeat([]byte("m"), 32)
+	err := mt.UpdateWord(
+		zkt.NewByte32FromBytesPaddingZero(key1),
+		zkt.NewByte32FromBytesPaddingZero(bytes.Repeat([]byte("v"), 32)),
+	)
+	assert.NoError(t, err)
+	err = mt.UpdateWord(
+		zkt.NewByte32FromBytesPaddingZero(key2),
+		zkt.NewByte32FromBytesPaddingZero(bytes.Repeat([]byte("n"), 32)),
+	)
+	assert.NoError(t, err)
+
+	proof := memorydb.New()
+	s_key1, err := zkt.ToSecureKeyBytes(key1)
+	assert.NoError(t, err)
+
+	sibling1, err := tr.ProveWithDeletion(s_key1.Bytes(), 0, proof)
+	assert.NoError(t, err)
+	nd, err := tr.TryGet(key2)
+	assert.NoError(t, err)
+	l := len(sibling1)
+	assert.Equal(t, sibling1[l-33:l-1], nd)
```

Comment:
Can you add an explanation for `[l-33:l-1]` here?

Reply:
Yes I just picked the value part (32 bytes) directly from the bytes of marshalling of leaf node, somewhat hacking tech. for we knew the encoding schema of leaf node. So as to compare it with the leaf value obtained from `TryGet` method.

I would add comment about it